src/server_manager/web_app/app.ts

```diff
@@ -855,42 +858,82 @@ export class App {
         });
   }
 
-  private async setAccessKeyDataLimit(limit: server.DataLimit) {
+  private async setDefaultDataLimit(limit: server.DataLimit) {
     if (!limit) {
       return;
     }
-    const previousLimit = this.selectedServer.getAccessKeyDataLimit();
+    const previousLimit = this.selectedServer.getDefaultDataLimit();
     if (previousLimit && limit.bytes === previousLimit.bytes) {
       return;
     }
     const serverView = this.appRoot.getServerView(this.appRoot.selectedServerId);
     try {
-      await this.selectedServer.setAccessKeyDataLimit(limit);
+      await this.selectedServer.setDefaultDataLimit(limit);
       this.appRoot.showNotification(this.appRoot.localize('saved'));
-      serverView.accessKeyDataLimit = dataLimitToDisplayDataAmount(limit);
+      serverView.defaultDataLimitBytes = limit?.bytes;
+      serverView.isDefaultDataLimitEnabled = !!limit;
```

Comment:
This is always true because you return early if the limit is absent.
```suggestion
      serverView.isDefaultDataLimitEnabled = true;
```